code-dot-org/code-dot-org
# [Bad Guys] Support for sharing and publishing

Builds on the work started in:
- https://github.com/code-dot-org/code-dot-org/pull/45449
Alternative to:
- https://github.com/code-dot-org/code-dot-org/pull/45468

At a high level, this PR adds support for the Bad Guys project type. Users will be able to create new projects which can be shared, remixed, published and viewed. The Bad Guys projects are in fact Sprite Lab projects (using a separate level), and are categorized as such in the gallery.

### Engineering Work
* Designate that `New The Bad Guys.level` is a project level.
* Create a project type for The Bad Guys
* Categorize `'spritelab'` and `'thebadguys'` projects both as "Sprite Lab"
* Add translation strings and thumbnail.
* Ensure the project shows in the Create dropdown correctly (it will not be a default there, but will show when working on a Bad Guys project)

*Public gallery showing a variety of projects. The Bad Guys projects are categorized as Sprite Lab:*
<img width="501" alt="image" src="https://user-images.githubusercontent.com/43474485/160637221-c6ea1ee7-414a-47fc-940f-b8b605ada354.png">

*Viewing all Sprite Lab projects...*
<img width="506" alt="image" src="https://user-images.githubusercontent.com/43474485/160637415-ef31b550-4149-457c-b0e5-2d1f46640983.png">

*Bad Guys option showing in the create menu when on a Bad Guy project:*
![image](https://user-images.githubusercontent.com/43474485/160644531-5fd89e15-8b81-43be-bacd-24a5aaaee4c3.png)

*Bad Guys option NOT showing in the create menu when on a different project:*
![image](https://user-images.githubusercontent.com/43474485/160644645-18a4fcaf-7fcd-4fdb-8d67-8c72405daa2c.png)


## Links
Jira Issue:
- [STAR-2164: [Bad Guys] Create New Level Type for Bad Guys](https://codedotorg.atlassian.net/browse/STAR-2164)


## Follow-up Work
* [STAR-2188: [Bad Guys] Final polish on freeplay level ](https://codedotorg.atlassian.net/browse/STAR-2188)
Mike will be making any final adjustments to the level, including any additional assets or changes to blocks and instructions.
* [STAR-2192: [Bad Guys] Enable The Bad Guys Project in /Projects](https://codedotorg.atlassian.net/browse/STAR-2192)
Assigned to Mike. After this PR merges, this should be a single-line code change that can be merged just ahead of the campaign launch. ([`StartNewProject.jsx`](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/projects/StartNewProject.jsx))
* [STAR-2190: [Bad Guys] Enable marketing changes](https://codedotorg.atlassian.net/browse/STAR-2190)
Kelby will be working to update the homepage, create a landing page, and update the Projects page - Brendan will be assisting and helping ramp up Kelby
* [STAR-2187: [Bad Guys] DCDO flip live 4/11](https://codedotorg.atlassian.net/browse/STAR-2187)
Brendan will coordinate timing with Marketing.

## Local Testing Instructions
* Create and publish a The Bad Guy project at http://localhost-studio.code.org:3000/projects/thebadguys
* Confirm that The Bad Guys button shows up correctly under the Create menu, but only from within that project.
* Look for the new project in the Sprite Lab public gallery at http://localhost-studio.code.org:3000/projects/
